adapters/oidc/js/src/main/resources/keycloak.js

```diff
@@ -873,10 +865,11 @@
         function checkLoginIframe() {
             var promise = createPromise();
 
-            if (loginIframe.iframe && loginIframe.iframeOrigin) {
+            if (loginIframe.iframe && loginIframe.iframeOrigin && loginIframe.callbackList.length !== 0) {
                 var msg = {};
                 msg.callbackId = createCallbackId();
-                loginIframe.callbackMap[msg.callbackId] = promise;
+                msg.sessionId = kc.sessionId;
+                loginIframe.callbackList.push(promise);
```

Comment:
If callbackList.length > 0 it needs to add promise to the callbackList (but don't postMessage). Othewise the promise won't be resolved by the message callback. Other than that it looks good. I'll test it once you've made the change.